lib/spack/spack/cray_manifest.py

```diff
@@ -132,7 +132,7 @@ def spec_from_entry(entry):
         variant_strs = list()
         for name, value in entry["parameters"].items():
             # TODO: also ensure that the variant value is valid?
-            if not (name in pkg_cls.variants):
+            if not (name in pkg_cls.variant_names()):
```

Comment:
Isn't this another way of saying:
```suggestion
            if not pkg_cls.has_variant(name):
```
?

Reply:
yep